### ISSUE TITLE
Add Logger Support

### ISSUE THREAD

**Issue by kdavisk6** [OpenFeign/feign-reactive#11]
The current implementation is locked to SLF4J.  We replace this with support for the `Logger` abstraction.

**Comment by kptfh** [comment]
Do you mean `feign.Logger`?

**Comment by kdavisk6** [comment]
Yes.

**Comment by kptfh** [comment]
'feign.Logger' is not designed for reactive work. Can introduce ReactorLogger but don't have enough reason for this as SLF4J is also a high level abstraction.

**Comment by kdavisk6** [comment]
I disagree, and we shouldn't ask users of OpenFeign to throw away concepts that are prevalent across all of our other integrations.  There are ways to take a synchronous process and make it reactive.  Take the reactor documentation itself, [B.1. How do I wrap a synchronous, blocking call?](http://projectreactor.io/docs/core/release/reference/#faq.wrap-blocking).  

Feign favor Composition over Inheritance.  It is reasonable for us to be able add support for both a native Reactive approach and an approach that allows users to use existing Synchronous Feign constructs.  By doing so we can increase the likely hood of this projects adoption.
